sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -56,6 +71,26 @@ trait Predicate extends Expression {
   override def dataType: DataType = BooleanType
 }
 
+/**
+ * The factory object for `BasePredicate`.
+ */
+object Predicate extends CodeGeneratorWithInterpretedFallback[Expression, BasePredicate] {
```

Comment:
do we still need `InterpretedPredicate` for creating interpreted Predicate specifically?

Reply:
Ur, I see. too verbose, so I'll remove it. Thanks!